x-pack/plugins/security_solution/public/management/components/endpoint_response_actions_list/components/use_action_history_url_params.ts

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+import {
+  RESPONSE_ACTION_COMMANDS,
+  RESPONSE_ACTION_STATUS,
+  type ResponseActions,
+  type ResponseActionStatus,
+} from '../../../../../common/endpoint/service/response_actions/constants';
+import { useUrlParams } from '../../../hooks/use_url_params';
+
+interface UrlParamsActionsLogFilters {
+  commands: string;
+  hosts: string;
+  statuses: string;
+  startDate: string;
+  endDate: string;
+  users: string;
+}
+
+interface ActionsLogFiltersFromUrlParams {
+  commands?: Array<
+    Exclude<ResponseActions, 'unisolate' | 'running-processes'> | 'release' | 'processes'
+  >;
+  hosts?: string[];
+  statuses?: ResponseActionStatus[];
+  startDate?: string;
+  endDate?: string;
+  setUrlActionsFilters: (commands: UrlParamsActionsLogFilters['commands']) => void;
+  setUrlDateRangeFilters: ({ startDate, endDate }: { startDate: string; endDate: string }) => void;
+  setUrlHostsFilters: (agentIds: UrlParamsActionsLogFilters['hosts']) => void;
+  setUrlStatusesFilters: (statuses: UrlParamsActionsLogFilters['statuses']) => void;
+  setUrlUsersFilters: (users: UrlParamsActionsLogFilters['users']) => void;
+  users?: string[];
+}
+
+type FiltersFromUrl = Pick<
+  ActionsLogFiltersFromUrlParams,
+  'commands' | 'hosts' | 'statuses' | 'users' | 'startDate' | 'endDate'
+>;
+
+export const actionsLogFiltersFromUrlParams = (
+  urlParams: Partial<UrlParamsActionsLogFilters>
+): FiltersFromUrl => {
+  const actionsLogFilters: FiltersFromUrl = {
+    commands: [],
+    hosts: [],
+    statuses: [],
+    startDate: 'now-24h/h',
+    endDate: 'now',
+    users: [],
+  };
+
+  const urlCommands = String(urlParams.commands)
+    .split(',')
+    .reduce<Required<ActionsLogFiltersFromUrlParams>['commands']>((acc, curr) => {
+      if (
+        RESPONSE_ACTION_COMMANDS.includes(curr as ResponseActions) ||
+        curr === 'release' ||
+        curr === 'processes'
+      ) {
+        acc.push(curr as Required<ActionsLogFiltersFromUrlParams>['commands'][number]);
+      }
+      return acc.sort();
+    }, []);
+
+  const urlHosts = urlParams.hosts ? String(urlParams.hosts).split(',').sort() : [];
```

Comment:
This (and `urlUsers` below) are the only ones doing a check first to ensure `urlParams.hosts` is thruthy. Should the others also make this check?

Reply:
Fair point. Will change.